addons/mail/models/mail_message.py

```diff
@@ -96,12 +96,12 @@ def default_get(self, fields):
         help='Attachments are linked to a document through model / res_id and to the message '
              'through this field.')
     parent_id = fields.Many2one(
-        'mail.message', 'Parent Message', index=True, ondelete='set null',
+        'mail.message', 'Parent Message', index="not null", ondelete='set null',
         help="Initial thread message.")
     child_ids = fields.One2many('mail.message', 'parent_id', 'Child Messages')
     # related document
-    model = fields.Char('Related Document Model', index=True)
-    res_id = fields.Many2oneReference('Related Document ID', index=True, model_field='model')
+    model = fields.Char('Related Document Model', index='not null')
```

Comment:
By the way, most messages have model / res_id set. Not sure why we need a "not null" as according to your message, it is necessary when most data is null ?

Reply:
done